tests/std/tests/P2442R1_views_chunk/test.cpp

```diff
@@ -197,23 +197,30 @@ constexpr bool test_one(Rng&& rng, Expected&& expected) {
         }
     }
 
-    STATIC_ASSERT(CanMemberEnd<const R> == forward_range<const V>);
+    STATIC_ASSERT(CanMemberEnd<const R> == input_range<const V>);
     if constexpr (CanMemberEnd<const R>) {
-        const same_as<sentinel_t<const R>> auto cs = as_const(r).end();
-        assert((r.begin() == cs) == is_empty);
-        STATIC_ASSERT(
-            common_range<R> == (common_range<const V> && (sized_range<const V> || !bidirectional_range<const V>) ));
-        if constexpr (common_range<const R> && bidirectional_range<V>) {
-            if (!is_empty) {
-                assert(equal(*prev(cs), *prev(end(expected))));
-            }
-
-            if constexpr (copy_constructible<V>) {
-                const auto r2 = r;
+        if constexpr (CanMemberBegin<const R>) {
+            const same_as<sentinel_t<const R>> auto cs = as_const(r).end();
+            assert((r.begin() == cs) == is_empty);
+            // clang-format off
+            STATIC_ASSERT(common_range<const R>

```

Comment:
Note the drive-by fix for the pre-existing bug: this was not validating `common_range<const R>`, but checking that `common_range<R>` is true when the conditions necessary for `common_range<const R>` hold. (@miscco, please check my math.) (No change requested.)

Reply:
Yeah, it seems that one slipped through. Thanks for fixing